### ISSUE TITLE
Make app build and run using Xcode 9 (requires upgrading realm)

### ISSUE THREAD

**Issue by oskarth** [status-im/status-react#1923]
As a developer I want to use the latest version of Xcode to build and run the app. Additionally, I want to stay up to date with realm dependency.

### Description

Using Xcode 9 the app currently doesn't build due to semantic issues in Realm. This has been fixed in later versions of Realm. Currently we have a `^0.14.3` dependency of realm which means we have missed out on the whole major 1 version for a while. Considering our use of `^` (autobump everything but major version) and glancing at https://github.com/realm/realm-js/blob/master/CHANGELOG.md there doesn't appear to be any major changes. Specifically the jump to `1.X.X` doesn't note any breaking changes.

However, running the app with latest realm results in `[object RealmObject] is not ISeqable` error.

### Acceptance criteria

Building Status with https://wiki.status.im/Building_Status or similar instructions should work on Xcode major version 9 with no regressions in app behavior.

*Type*: Bug

Operating System: iOS

**Comment by oskarth** [comment]
Also seeing this:

```
Oskars-MacBook-Pro:~ oskarth$ cd "/Users/oskarth/git/status-react"; tail -f '/Users/oskarth/Library/Logs/CoreSimulator/3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7/system.log'
Sep 25 14:05:14 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.uifoundation-bundle-helper): Unknown key for Boolean: EnableTransactions
Sep 25 14:05:14 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.uifoundation-bundle-helper): Unknown key for Boolean: EnablePressuredExit
Sep 25 14:05:14 Oskars-MacBook-Pro budd[65968]: objc[65968]: Class BYBuddyDaemonCloudSyncClient is implemented in both /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/SetupAssistant.framework/SetupAssistant (0x108304820) and /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/SetupAssistant.framework/budd (0x107ded1f8). One of the two will be used. Which one is undefined.
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.DictionaryServiceHelper): Unknown key for Boolean: EnableTransactions
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.DictionaryServiceHelper): Unknown key for Boolean: EnablePressuredExit
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.StreamingUnzipService): Unknown key for Boolean: EnableTransactions
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.StreamingUnzipService): Unknown key for Boolean: EnablePressuredExit
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.uifoundation-bundle-helper): Unknown key for Boolean: EnableTransactions
Sep 25 14:05:18 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.uifoundation-bundle-helper): Unknown key for Boolean: EnablePressuredExit
Sep 25 14:05:19 Oskars-MacBook-Pro com.apple.CoreSimulator.SimDevice.3DFE45B4-574F-4DCB-860B-2FB5CFB22BD7[65940] (com.apple.videosubscriptionsd[66244]): Service exited with abnormal code: 1
Sep 25 14:06:44 Oskars-MacBook-Pro StatusIm[66274]: assertion failed: 16G29 15A372: libxpc.dylib + 69578 [D870A237-D3A7-31F5-AAD4-CE880C0C8E7B]: 0x7d
Sep 25 14:07:00 Oskars-MacBook-Pro StatusIm[66274]: DEPRECATED USE in libdispatch client: Setting timer interval to 0 requests a 1ns timer, did you mean FOREVER (a one-shot timer)?
```

UPDATE: This seems to be a problem with realm upgrade, see that issue. I don't know if there are more issues or not with Xcode9 upgrade.

**Comment by oskarth** [comment]
Updated user story: As a developer, I want to get on with my life and fix core problems we are having, not spend time wrestling with a less than week old version of Xcode that's more like a release candidates and Apple pushed on me behind my back.

Attempting to downgrade to 8.3.3. Hopefully some Xcode 9 issues will resolve themselves in coming weeks. Realm 1.X upgrade has to be done beforehand regardless but possible this is more straightforward using Xcode 8.X.X. Will create separate issue for this as a blocker for this one.

**Comment by oskarth** [comment]
To toggle versions of xcode, you can download previous version here https://developer.apple.com/download/more/ then rename `app` to `Xcode8` and `Xcode9` then run: `sudo xcode-select --switch /Applications/Xcode8.app/`. Could be useful for the person who ends up taking this issue.

**Comment by goranjovic** [comment]
I am getting simulator version conflicts even with 8.3.3

**Comment by oskarth** [comment]
@goranjovic Could you elaborate? if you mean things like 14->13 error or printing wrong schema in logs, this is likely because of switching branches. Easiest to fix is to uninstall app, use a different device or reset contents for whole simulator device.

**Comment by hitripod** [comment]
+1 The Same issue here.

**Comment by oskarth** [comment]
@janherich Please write here what error you got when you last attempted to run this as it is very relevant info for person taking this issue.

**Comment by janherich** [comment]
@oskarth I'm developing currently with latest XCode 9.1 and as far as running the dev setup (iOS simulator + figwheel + re-frisk) everything works wonderfully, the only problem is opening the project file in XCode and trying to build it there, when I tried that, this is the error I got:
![screen shot 2017-11-19 at 00 31 14](https://user-images.githubusercontent.com/1910609/33035608-0f786c82-ce2c-11e7-9ca9-85260610f9dd.png)

**Comment by oskarth** [comment]
Seems like a linking step has changed, should be fixable by clicking around in Xcode a bunch https://stackoverflow.com/questions/9863836/library-not-found-for-lpods (this type of problem is fairly common). Main uncertainty if this is only step left or if it is step 2 of N where N is large.

**Comment by annadanchenko** [comment]
after discussion with a team, not adding bounty to the issue yet: `I would say not now, first react-native update on OSX and if it doesn't work on xcode 9 after that (but it probably will) we can think about it`

**Comment by rasom** [comment]
resolved in `develop`
